include/net/buf.h

```diff
@@ -818,7 +818,7 @@ struct net_buf_pool {
 
```

Comment:
Thanks Ehud, looks good. Couple of nits:
* change `atomic_add()` to `atomic_inc()`
* please change commit subject to `net: buf: change avail_count variable to atomic type`